echopype/clean/utils.py

```diff
@@ -6,8 +10,29 @@
 from ..utils.compute import _lin2log, _log2lin
 
 
-def calc_transient_noise_pooled_Sv(
-    ds_Sv: xr.Dataset, func: str, depth_bin: float, num_side_pings: int, exclude_above: float
+def extract_dB(dB_str: str) -> float:
+    """Extract float value from decibal string in the form of 'NUMdB'."""
+    # Search for numeric part using regular expressions and convert to float
+    if not isinstance(dB_str, str):
+        raise TypeError(
+            "Decibal input must be a string formatted as `NUMdB` or `NUMdb."
+            f"Cannot be of type `{type(dB_str)}`."
+        )
+    pattern = r"^[-+]?\d+\.?\d*(?:dB|db)$"  # Ensures exact match with format
```

Comment:
do you want to add an optional whitespace before dB/db so that inputs could be `NUMdB` or `NUM dB`?

Reply:
I don't like having this space here. Maybe we just keep it `NUMdB` without space? This way it's also simpler and the user only has 2 accepted ways to write it, `NUMdB` and `NUMdb`, instead of 4 ways to write it

Reply:
i think this is fine. I just remembered that we also only allow "10m".